lib/active_model/serializer/associations.rb

```diff
@@ -92,7 +92,7 @@ def associations(include_tree = DEFAULT_INCLUDE_TREE)
 
         Enumerator.new do |y|
           self.class._reflections.each do |reflection|
-            next unless include_tree.key?(reflection.name)
+            next unless include_tree.try(:key?, reflection.name)
```

Comment:
Clearly, the test should be on `reflection.key` instead of `reflection.name` in order to handle custom keys (`reflection.key` falls back to `reflection.name` anyways if one was not provided). I'm still trying to understand what caused `include_tree` to be `nil` though.